src/js/plugins/plugin.text_selection.js

```diff
@@ -0,0 +1,204 @@
+//@ts-check
+import { isFirefox } from './tts/utils';
+
+const BookReader = /** @type {typeof import('../BookReader').default} */(window.BookReader);
+
+class TextSelectionPlugin {
```

Comment:
Let's export this for testing

```suggestion
export class TextSelectionPlugin {
```

Reply:
done